omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -110,6 +112,28 @@ def from_file(filename)
       end
     end
 
+
+    VALID_EXTENSION_CONFIGS= %i(server_config_required config_values gen_backend gen_frontend gen_secrets gen_api_fqdn) unless defined?(VALID_EXTENSION_CONFIGS)
+    def register_extension(name, extension)
+      bad_keys = extension.keys - VALID_EXTENSION_CONFIGS
+
+      bad_keys.each do |key|
+        Chef::Log.warn("Extension #{name} contains unknown configuration option: #{key}")
+        extension.delete(key) # delete mutates extension
+      end
+
+      PrivateChef["registered_extensions"][name] = extension
+      if extension.key?(:config_values)
+        extension[:config_values].each do |k, v|
+          # TODO(ssd): Should we even allow this? Perhaps we should just exit?
+          if PrivateChef.has_key?(k)
+            Chef::Log.warn("Extension #{name} attempted to register configuration default for #{k}, but it already exists!")
+          end
+          PrivateChef.send(k, v)
```

Comment:
Could we instead use `PrivateChef[k] = v`? I had to think this through for a few seconds and reember what PrivateChef inherited from to realize what this would do. 

We also recently ran into an issue where we had a function sharing a name with a key, and it made serverspec very sad.  Maybe nest it one level under PrivateChef['config_extensions'][k] to avoid trouble?